### DISCUSSION TITLE
RenderTargetBitmap not rendering ToggleSwitch

### DISCUSSION THREAD

**Discussion by JBarbas** [AvaloniaUI/Avalonia#14316] [Q&A]
I am rendering images from a canvas with different controls within (as children). As far as I tested everything has worked fine except the ToggleSwitch, which does not appear in the renders.

For instance I have this canvas:

```
<Canvas>
  <PathIcon Data="{StaticResource engine}"/>
  <PathIcon Data="{StaticResource power}"/>
  <TextBlock Text="Combustion:"/>
  <TextBlock Text="Electric:"/>
  <ToggleSwitch IsChecked="True"/>
  <ToggleSwitch IsChecked="False"/>
</Canvas>
```

Which displays this:
![View](https://github.com/AvaloniaUI/Avalonia/assets/26795644/b5ee6388-5b8b-4c5f-93cf-473bbb6263d3)

But when I render it, shows the following:
![Render (1)](https://github.com/AvaloniaUI/Avalonia/assets/26795644/ac5ac2aa-3b6e-417e-91ac-a3b1cd00f0a0)

To render the bitmap I am using this code:
```
void RenderToFile(Control target, string path)
{
    var pixelSize = new PixelSize((int) target.Width, (int) target.Height);
    var size = new Size(target.Width, target.Height);
    using (RenderTargetBitmap bitmap = new RenderTargetBitmap(pixelSize, new Vector(96, 96)))
    {
        target.Measure(size);
        target.Arrange(new Rect(size));
        bitmap.Render(target);
        bitmap.Save(path);
    }
}
```
Which I got from here: https://github.com/AvaloniaUI/Avalonia.HtmlRenderer/issues/10

I have tested it on the following versions:

- 11.0.4
- 11.0.7
- 11.0.999 (last nightly stable version)

**Comment by timunie** [comment]
Use Bounds instead of Width / Height. This will get current size while Width and Height is undefined in your case. Bounds is equivalent to WPFs `ActualWidth / ActualHeight`

```diff
void RenderToFile(Control target, string path)
{
!	var pixelSize = new PixelSize((int) target.Bounds.Width, (int) target.Bounds.Height);
!	var size = target.Bounds.Size; 
	
	using (RenderTargetBitmap bitmap = new RenderTargetBitmap(pixelSize, new Vector(96, 96)))
	{
		target.Measure(size);
		target.Arrange(new Rect(size));
		bitmap.Render(target);
		bitmap.Save("Test.png");
	}
}
```

**Reply by maxkatz6** [reply]
Also, would make sense to get Bounds after Measure/Arrange.
